engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java

```diff
@@ -67,8 +67,9 @@ public Warning[] getWarningsArray(WarningsLibrary warningsLibrary) {
     return allWarnings;
   }
 
-  public ArrayRope<Warning> collectWarnings() {
-    return warnings;
+  /** @return the number of warnings. */
+  public int getWarningsCount() {
```

Comment:
More of encapsulation (e.g. returning just number instead whole data structure)? I always like being more conservative and encapsulate more.